### ISSUE TITLE
RFC: Remove overview section, add reference to front page of http://hacklang.org/ instead

### ISSUE THREAD

**Issue by fredemmott** [facebook/hhvm#6697]
I think the content and phrasing of hacklang.org is much better for quickly getting someone interested in Hack:

Specifically:

 - the introduction section is just an extra click (removed already but not pushed)
 - the typing section doesn't make it clear what features Hack has that PHP5 or 7 don't - https://docs.hhvm.com/hack/overview/typing
 - the async page is just an abridged version of /hack/async/introduction - https://docs.hhvm.com/hack/overview/async
 - the collections section is very long, and doesn't convey much more information than hacklang.org's single paragraph - https://docs.hhvm.com/hack/overview/collections

cc @JoelMarcey  @simonwelsh @matthewjohnston4 @SiebelsTim  @jazzdan  to get a variety of perspectives

**Comment by SiebelsTim** [comment]
@fredemmott was this supposed to be in user-documentation?

I do like the async page, as it's much shorter than async/introduction and still gives away the idea.

When I'm reading docs I really like to see code. That's an advantage for the docs collection overview in contrast to hacklang.org. 

---

I really like the typechecked box at hacklang and the brevity of the page.
The Doc's overview gives a more detailed view and though.

Bottom line, I think you'd be better at the docs if you're interested already, and hacklang.org is better advertisement.

**Comment by fredemmott** [comment]
Whoops, yes, wrong repo. Will keep it here for now instead of copying over.

So... if we keep the section, I think it should cover a lot more topics: I'm also a little worried that someone could look through it and think that's all we have to offer.

**Comment by jazzdan** [comment]
I think having some sort of overview as part of the docs is valuable, and I agree that the "typing", "async" and "collections" sections of the overview seem a bit superfluous.

As the PHP community moves to PHP 7, is the emphasis on type annotations worthwhile? I think that reframing it as "what extra typing features does Hack add on top of PHP 7's type features" makes a lot of sense.

As I'm thinking through this, I'm coming around to the idea that hacklang.org should just be the introduction. The overview I would like to see is a list of Hack's biggest features, with simple examples or explanations to demonstrate their utility. This seems to be exactly what hacklang.org is. :)

**Comment by JoelMarcey** [comment]
Wanted to get this back to the user-documentation repo. So I created an issue there.
